src/cli/cli.cpp

```diff
@@ -1453,6 +1453,20 @@ template <> otError Interpreter::Process<Cmd("delaytimermin")>(Arg aArgs[])
 }
 #endif
 
+template <> otError Interpreter::Process<Cmd("detach")>(Arg aArgs[])
+{
+    otError error = OT_ERROR_NONE;
+
+    SuccessOrExit(error = otThreadDetachGracefully(GetInstancePtr(), &Interpreter::HandleDetachGracefullyResult, this));
+
+    mDetachAsync = (aArgs[0] == "async");
+
+    error = mDetachAsync ? OT_ERROR_NONE : OT_ERROR_PENDING;
```

Comment:
In async case, can we just not register a callback (pass in `nullptr`)?
This way I think we won't need `mDetachAync` variable anymore?

Reply:
I'm using the callback to print a message to indicate the completion, but I think removing the message is fine as it's a fast operation. SGTM.

Reply:
Done.